### PR TITLE
SQL: adds format parameter to range queries for constant date comparisons

### LINKED ISSUE #45139: SQL: conditions involving date fields with custom format and SQL-generated dates fail
For a `date` field with a custom format like `HH:mm yyyy-MM-dd`, a simple query like `SELECT * FROM test WHERE date > NOW()` will fail with an error:

```
        "root_cause": [
            {
                "type": "parse_exception",
                "reason": "failed to parse date field [2019-08-02T10:42:15.773Z] with format [HH:mm yyyy-MM-dd]: [Text '2019-08-02T10:42:15.773Z' could not be parsed at index 2]"
            }
        ],
```

Moreover, if a timezone is passed in the request in the {area}/{city} format (for REST - for example `"time_zone": "Europe/Paris"`), the date SQL generates and places it in the `range` filter also specifies the {area}/{city} format alongside the offset. The offset should be enough and the area/city is redundant.

```
        "root_cause": [
            {
                "type": "parse_exception",
                "reason": "failed to parse date field [2019-08-02T12:42:53.804+02:00[Europe/Paris]] with format [HH:mm yyyy-MM-dd]: [Text '2019-08-02T12:42:53.804+02:00[Europe/Paris]' could not be parsed at index 2]"
            }
        ],
```

### PR DESCRIPTION
**Pull request by astefan**
Whenever a date field is used in a <, >, <=, >= comparison, a `range` query is built. This includes any output from `CURRENT_*`/`NOW`/`TODAY` functions. This PR adds the `format` parameter to these range queries and sets the value of the date constant as String.

Also, it changes the = and != operators to use a `range` query as well, to have the proper format for the literal used in the equality/non-equality condition.

Fixes https://github.com/elastic/elasticsearch/issues/45139.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/default-distro

File: x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/CustomDateFormatTestCase.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+        index.endObject();
+        index.endObject();
+
+        request.setJsonEntity(Strings.toString(index));
+        client().performRequest(request);
+    }
+    
+    private ZonedDateTime nowWithMillisResolution(String zoneId) {
```
Comment (costin) [inline x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/CustomDateFormatTestCase.java]: Can't you use `DateUtils.nowWithMillisResolution`?

**Review by costin — APPROVED** [review]
LGTM

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Pull request merged**